quality-assurance/core.md

```diff
@@ -0,0 +1,194 @@
+    *   [Blackbox](#blackbox)  
+    *   [Whitebox](#whitebox)  
+*   [Processes](#processes)
+    *   [Automated Testing](#automated-testing)
+    *   [Manual Testing](#manual-testing)
+
+## What makes a good web application?
+The following is a checklist of things to think about both during development of a product and during the quality 
+assurance phase.
+
+### Blackbox
+**General**
+* All UI scales well and looks good on small form factor devices
+* All errors are handled and UI reacts appropriately
+* Errors are clear in meaning &amp; reflect what happened
+* Activity indicators are used when loading data asynchronously
+* Empty information is handled correctly in UI (optional fields etc)
+* Long running operations give a indication of progress if possible
+* Breadcrumbs are used where appropriate
+
+**Navbar**
+* UI reflects the current active page
+* Icon is appropriate for the page
+
+**Forms**
+* Invalid inputs correctly trigger validation errors
+* All form inputs have associated labels
+* Appropriate validation for different inputs (phone number, postcode)
+* The forms are not too long &amp; and are grouped logically
+* The user is alerted of the form lifecycle (success, error)
+* When modals are used in the context of updates, ensure inputs are correctly pre populated.
+* Autofill is disabled on inputs where appropriate
+* Inputs have correct type property ( emails have email type, phone numbers have tel type)
+
+**Modals**
+* Modals are animated on both display and dismiss
+* Modals are used where appropriate over pages
+
+**Formatting**
+* Dates are formatted correctly for the appropriate audience.
+* Monetary values are formatted correctly for the appropriate audience
+* Phone numbers are formatted correctly for the appropriate audience
+
+**Pagination**
+* UI correctly indicates current page
+* UI shows at least 2 page tabs ahead and below the current page
+
+**Login &amp; Auth**
+* User can only access pages that require 
+* Authentication when they are logged in
+* Login &amp; Logout works as expected
+* Reset password works as expected
+
+**Registration**
+* Email addresses are verified &amp; unique (if applicable)
+* Passwords meet minimum standards (min 8 chars)
+* Passwords should not have maximum length restrictions (allow at least 255 characters).
+
+**Emails**
+* Emails look good on small form factor devices
+* Ensure correct subject &amp; sender name
+* Correct email is sent, information is conveyed appropriately
+* Emails are received in a timely manner
+* Emails come from the appropriate email (often the clients)
+* Links point to the correct address
+
+**Push Notifications**
+* Push notifications are received in a timely manner
+* They show correct &amp; relevant information
+
+**Permissions Roles**
+* Ensure permissions &amp; roles match up to possible actions (only admins can create users etc)
+
+### Whitebox
+**General**
+* Filenames must follow standard coding conventions (UsersController.php)
```

Comment:
```suggestion
* Filenames must follow standard coding conventions (UserController.php)
```

I prefer to have singular controller names, to continue the naming convention of having singular models

Reply:
It's all opinion based, I prefer the plural but it looks like on the Laravel docs they use singular naming so that makes sense.